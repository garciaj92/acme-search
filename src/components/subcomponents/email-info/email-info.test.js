import React from "react";
import EmailInfo from "./email-info";

import { mount } from "enzyme";
import toJson from 'enzyme-to-json';

const props = {
    emails: ["test1@test.com, test2@test.com"],
};

describe("EmailInfo: Test visual components", () => {
    let wrapper;
    beforeEach(() => {
        wrapper = mount(<EmailInfo {...props} />);
    });

    it("renders correctly", () => {
        expect(toJson(wrapper)).toMatchSnapshot();
    })

    // it("displays correct title and date", () => {
    //     expect(wrapper.find("h5").first().text()).toEqual(props.title);
    //     expect(wrapper.find("p").first().text()).toEqual(props.date);
    // });
});

// describe("EmailInfo: Test prop type validation", () => {
//     let consoleErrorSpy;
//     beforeEach(() => {
//         console.error = jest.fn()
//         consoleErrorSpy = jest.spyOn(console, "error");
//     });

//     afterEach(() => {
//         // It's best practice to clear spied function after every test run.
//         // https://stackoverflow.com/questions/43245040/
//         //   using-jest-to-spy-on-method-call-in-componentdidmount/43284406#43284406
//         consoleErrorSpy.mockClear();
//     });
    
//     it("throws an error when title prop is missing", () => {
//         mount(<EmailInfo date={props.date} />);
//         expect(consoleErrorSpy).toHaveBeenCalledTimes(1);
//     });
  
//     it("throws an error when date prop is missing", () => {
//         mount(<EmailInfo title={props.title} />);
//         expect(consoleErrorSpy).toHaveBeenCalledTimes(1);
//     });
// });
