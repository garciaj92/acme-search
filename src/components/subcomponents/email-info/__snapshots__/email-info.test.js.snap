// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmailInfo: Test visual components renders correctly 1`] = `
<EmailInfo
  emails={
    Array [
      "test1@test.com, test2@test.com",
    ]
  }
>
  <div
    className="d-sm-inline-block align-baseline-top m-3"
  >
    <CardIconSubHeading
      heading="Email"
      iconSrc="email.svg"
    >
      <div
        className="d-flex justify-content-center align-items-center my-2"
      >
        <Image
          fluid={false}
          rounded={false}
          roundedCircle={false}
          src="email.svg"
          thumbnail={false}
          width={30}
        >
          <img
            className=""
            src="email.svg"
            width={30}
          />
        </Image>
        <h5
          className="mb-0 mx-1"
        >
          Email
        </h5>
      </div>
    </CardIconSubHeading>
    <CardLink
      content="test1@test.com, test2@test.com"
      href="mailto:test1@test.com, test2@test.com"
      key="0"
    >
      <a
        className="text-muted mb-0 d-block"
        href="mailto:test1@test.com, test2@test.com"
        onClick={[Function]}
      >
        test1@test.com, test2@test.com
      </a>
    </CardLink>
  </div>
</EmailInfo>
`;
